src/erc4626/ERC4626Oracle.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {IERC20} from "./IERC20.sol";
+import {IERC4626} from "./IERC4626.sol";
+
+contract ERC4626Oracle {
+    error OracleUnsupportedPair();
+
+    IERC4626 public immutable VAULT;
+    address public immutable ASSET;
+
+    constructor(address _vault) {
+        VAULT = IERC4626(_vault);
+        ASSET = VAULT.asset();
```

Comment:
```
VAULT_SCALAR = 10 ** VAULT.safeDecimals();
ASSET_SCALAR = 10 ** ASSET.safeDecimals();
```

They will come useful later.

Also, we should use `safeDecimals` for weird ERC20s